File: LeetCode/0646. Maximum Length of Pair Chain/Solution.kt
/**
 * 205 / 205 test cases passed.
 * Runtime: 244 ms
 * Memory Usage: 44.2 MB 
 */
class Solution {
    fun findLongestChain(pairs: Array<IntArray>): Int {
        var prev = Int.MIN_VALUE
        var ans = 0
        pairs.apply {
            sortBy { it[1] }
            forEach {
                if (it[0] > prev) {
                    prev = it[1]
                    ++ ans
                }
            }
        }
        return ans
    }
}

File: LeetCode/0646. Maximum Length of Pair Chain/solution.py
"""
205 / 205 test cases passed.
Runtime: 52 ms
Memory Usage: 15.2 MB
"""
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        prev, ans = -inf, 0
        for a, b in sorted(pairs, key=lambda p: p[1]):
            if a > prev:
                prev = b
                ans += 1
        return ans
